<?xml version="1.0"?>
<launch>
  <arg name="simulation" default="false" />
  <arg name="uav_odom_topic_name" default="uav_odom" />
  <arg name="image_name" default="/camera/image_rect" />

  <!-- vision base -->
  <include file="$(find jsk_uav_forest_perception)/launch/image_process.launch">
    <arg name="image_name" value="$(arg image_name)" unless="$(arg simulation)"/>
  </include>

  <!-- 2D laser base -->
  <include file="$(find jsk_uav_forest_perception)/launch/laser_process.launch">
  </include>

  <!-- tree detector -->
  <node pkg="jsk_uav_forest_perception" type="tree_tracking_node" output="screen" name="tree_tracking_node">
    <param name="laser_scan_topic_name" value="/scan_clustered" />
    <param name="uav_odom_topic_name" value="/ground_truth/state" if="$(arg simulation)"/>
    <param name="uav_odom_topic_name" value="$(arg uav_odom_topic_name)" unless="$(arg simulation)"/>
    <param name="filter_rate" value="0.8" />    
    <param name="uav_tilt_thre" value="0.1" />
    <param name="tree_margin_radius" value="1.0" />
    <param name="valid_num" value="15" />
    <param name="searching_method" value="2" /> <!-- 1:narrow 2:deep-->
    <param name="tree_radius_max" value="0.3" />
    <param name="tree_radius_min" value="0.07" />
    <param name="tree_scan_angle_thre" value="0.1" />
    <param name="tree_circle_regulation_thre" value="0.02" />
    <param name="first_tree_pos_margin" value="0.5" /> <!-- positive value -->
    <param name="first_tree_dist_thresh" value="3.0" /> <!-- positive value -->   
    <param name="narrow_searching_radius" value="8.0" />
    <param name="narrow_angle_diff_min" value="0.02" />
    <param name="max_orthogonal_dist" value="2.0" />
    <param name="only_target" value="false" />
    <param name="verbose" value="false" />
    <param name="visualization" value="true" />
  </node>
</launch>
